stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/TypeAdapterGenerator.java

```diff
@@ -268,8 +264,21 @@ private static MethodSpec getReadMethodSpec(@NotNull TypeName typeName,
                 }
             }
 
-            builder.addCode("\t\t\t\tobject." + variableName + " = " +
-                            adapterFieldInfo.getAdapterAccessor(elementValue) + ".read(reader);");
+            String variableType = element.getValue().toString();
+            boolean isPrimitive = TypeUtils.isSupportedPrimitive(variableType);
+
+            if(isPrimitive) {
+                builder.addCode("\t\t\t\t" + TypeUtils.getObjectForPrimitive(variableType) + " " + variableName + " = " +
+                        adapterFieldInfo.getAdapterAccessor(elementValue) + ".read(reader);\n");
+                builder.addCode("\t\t\t\tif(null != " + variableName + ") {\n");
+                builder.addCode("\t\t\t\t\tobject." + variableName + " = " + variableName + ";\n");
+                builder.addCode("\t\t\t\t}");
```

Comment:
I see this prevents us from getting an exception if the gson primitive type adapter returns null, that's great. See above for my comment about consistency between gson primitive type adapters and our type adapters